superdesk/data_updates/00007_20180321-092824_archive.py

```diff
@@ -0,0 +1,42 @@
+# -*- coding: utf-8; -*-
+# This file is part of Superdesk.
+# For the full copyright and license information, please see the
+# AUTHORS and LICENSE files distributed with this source code, or
+# at https://www.sourcefabric.org/superdesk/license
+#
+# Author  : tomas
+# Creation: 2018-03-21 09:28
+
+import os
+import subprocess
+import json
+from superdesk.commands.data_updates import DataUpdate
+
+def get_updated_editor_state(editor_state):
+    try:
+        p1 = subprocess.Popen(["echo", json.dumps(editor_state)], stdout=subprocess.PIPE)
```

Comment:
Don't use `echo` to send data, you have `communicate` method for that (which you use below), just pass your data to it.
Also you should use the `with` statement: 
```python
with subprocess.Popen(…) as p:
    p.do_something
```

Reply:
Fixed it :)